jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ManagedBeanDiagnosticsCollector.java

```diff
@@ -104,13 +102,12 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
                      * https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#managed_beans
                      */
                     if (isManagedBean && Flags.isPublic(fieldFlags) && !Flags.isStatic(fieldFlags)
-                            && managedBeanAnnotations.stream()
-                                    .anyMatch(annotation -> !annotation.equals("Dependent"))) {
-                        String DiagnosticMessage = createAnnotationDiagnostic("Dependent","be the only scope defined by a managed bean with a non-static public field.");
-                        Diagnostic diagnostic = createDiagnostic(field, unit,
-                                DiagnosticMessage,
-                                DIAGNOSTIC_CODE);
-                        diagnostics.add(diagnostic);
+                            && (fieldScopes.size() != 1 || !fieldScopes.get(0).equals(DEPENDENT_FQ_NAME))) {
+                        diagnostics.add(createDiagnostic(field, unit,
+                                createAnnotationDiagnostic("Dependent",
```

Comment:
We can use the constant: https://github.com/eclipse/lsp4jakarta/pull/318/files#diff-42b7f8b89fd1b493d43d4d94b32d983bba64ddfd042786875f6501a938375e35R32 rather than hardcoding "Dependent"